src/components/shared/SearchCard/SearchCardAction/RequestExperimentFormButton.js

```diff
@@ -1,12 +1,30 @@
 import { useModal } from 'hooks/useModal'
+import { useRefinebio } from 'hooks/useRefinebio'
 import { Button } from 'components/shared/Button'
+import { InlineMessage } from 'components/shared/InlineMessage'
 import { ModalPage } from 'components/shared/ModalPage'
 import { RequestExperimentForm } from './RequestExperimentForm'
 
 export const RequestExperimentFormButton = ({ accessionCode }) => {
+  const { requestedExperiments, setRequestedExperiments } = useRefinebio()
   const { closeModal, openModal } = useModal()
   const id = `request-experiment-form-${accessionCode}`
 
+  const addRequestedExperiment = () => {
+    setRequestedExperiments([...requestedExperiments, accessionCode])
+  }
```

Comment:
Can't we do this inside of the component? I think this should be defined in the context and exported.

Reply:
`addRequestedExperiment` is now defined inside the `RequestExperimentForm` component at 9035075f956fd9fe92334cc17a1243b16badcb57 👍